cmd/protoc-gen-openapi/generator/generator.go

```diff
@@ -817,6 +856,41 @@ func (g *OpenAPIv3Generator) addSchemasForMessagesToDocumentV3(d *v3.Document, m
 			contains(g.generatedSchemas, schemaName) {
 			continue
 		}
+		// Skip map entries and enums since they are handled separately
+		if len(message.Enums) > 0 {
+			for _, enum := range message.Enums {
+				enumName := string(enum.Desc.Name())
+				enumSchema := &v3.SchemaOrReference{
+					Oneof: &v3.SchemaOrReference_Schema{
+						Schema: &v3.Schema{
+							Type:   "string",
+							Format: "enum",
+							Enum: func() []*v3.Any {
+								enums := make([]*v3.Any, 0, enum.Desc.Values().Len())
+								for i := 0; i < enum.Desc.Values().Len(); i++ {
+									enums = append(enums, &v3.Any{
+										Yaml: string(enum.Desc.Values().Get(i).Name()),
+									})
+								}
+								return enums
+							}(),
```

Comment:
We can create the slice upfront rather than relying on an anonymous func here. Let's refactor later (if ever).